internal/impl/snowflake/streaming/rest.go

```diff
@@ -0,0 +1,478 @@
+	userName := strings.ToUpper(c.user)
+	issueAtTime := time.Now().UTC()
+	token := jwt.NewWithClaims(jwt.SigningMethodRS256, jwt.MapClaims{
+		"iss": fmt.Sprintf("%s.%s.%s", accountName, userName, "SHA256:"+base64.StdEncoding.EncodeToString(hash[:])),
+		"sub": fmt.Sprintf("%s.%s", accountName, userName),
+		"iat": issueAtTime.Unix(),
+		"exp": issueAtTime.Add(time.Hour).Unix(),
+	})
+	return token.SignedString(c.privateKey)
+}
+
+// RunSQL executes a series of SQL statements. It's expected that these statements execute in less than 45 seconds so
+// we don't have to handle async requests.
+func (c *SnowflakeRestClient) RunSQL(ctx context.Context, req RunSQLRequest) (resp RunSQLResponse, err error) {
+	requestID := uuid.NewString()
+	err = c.doPost(ctx, fmt.Sprintf("https://%s.snowflakecomputing.com/api/v2/statements?requestId=%s", c.account, requestID), req, &resp)
+	return
+}
+
+// configureClient configures a client for Snowpipe Streaming.
+func (c *SnowflakeRestClient) configureClient(ctx context.Context, req clientConfigureRequest) (resp clientConfigureResponse, err error) {
+	requestID := uuid.NewString()
+	err = c.doPost(ctx, fmt.Sprintf("https://%s.snowflakecomputing.com/v1/streaming/client/configure?requestId=%s", c.account, requestID), req, &resp)
+	return
+}
+
+// channelStatus returns the status of a given channel
+func (c *SnowflakeRestClient) channelStatus(ctx context.Context, req batchChannelStatusRequest) (resp batchChannelStatusResponse, err error) {
+	requestID := uuid.NewString()
+	err = c.doPost(ctx, fmt.Sprintf("https://%s.snowflakecomputing.com/v1/streaming/channels/status?requestId=%s", c.account, requestID), req, &resp)
+	return
+}
+
+// openChannel opens a channel for writing
+func (c *SnowflakeRestClient) openChannel(ctx context.Context, req openChannelRequest) (resp openChannelResponse, err error) {
+	requestID := uuid.NewString()
+	err = c.doPost(ctx, fmt.Sprintf("https://%s.snowflakecomputing.com/v1/streaming/channels/open?requestId=%s", c.account, requestID), req, &resp)
+	return
+}
+
+// dropChannel drops a channel when it's no longer in use.
+func (c *SnowflakeRestClient) dropChannel(ctx context.Context, req dropChannelRequest) (resp dropChannelResponse, err error) {
+	requestID := uuid.NewString()
+	err = c.doPost(ctx, fmt.Sprintf("https://%s.snowflakecomputing.com/v1/streaming/channels/drop?requestId=%s", c.account, requestID), req, &resp)
+	return
+}
+
+// registerBlob registers a blob in object storage to be ingested into Snowflake.
+func (c *SnowflakeRestClient) registerBlob(ctx context.Context, req registerBlobRequest) (resp registerBlobResponse, err error) {
+	requestID := uuid.NewString()
+	err = c.doPost(ctx, fmt.Sprintf("https://%s.snowflakecomputing.com/v1/streaming/channels/write/blobs?requestId=%s", c.account, requestID), req, &resp)
+	return
+}
+
+func (c *SnowflakeRestClient) doPost(ctx context.Context, url string, req any, resp any) error {
+	marshaller := json.Marshal
+	if debug {
```

Comment:
Ugh, we really should add an API for determining the log level. Task for another day :)

Reply:
Yupp. Another annoying thing - I could not figure out how to make a mock logger for tests, hence the `fmt.Println` stuff

Reply:
OK, no worries! We can figure that out some other time :)